From f74c3d2e3345726c18609f522f0e97b5361a8bb2 Mon Sep 17 00:00:00 2001
From: Florent Pirou <florent.pirou@intel.com>
Date: Wed, 19 Jun 2024 21:45:06 +0800
Subject: [PATCH 21/62] d4xx: add csi2 4 and 5 ports mapping, shift aggregated
 suffixes

Signed-off-by: Florent Pirou <florent.pirou@intel.com>
---
 drivers/media/i2c/d4xx.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index d5bc3097a..6a1b04dbe 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -3243,7 +3243,7 @@ static int ds5_board_setup(struct ds5 *state)
 		}
 	}
 	if (state->aggregated)
-		suffix += 4;
+		suffix += 6;
 	dev_info(dev, "Init SerDes %c on %d@0x%x<->%d@0x%x\n",
 		suffix,
 		bus, pdata->subdev_info[0].board_info.addr, //48
@@ -3752,7 +3752,7 @@ static int ds5_sensor_init(struct i2c_client *c, struct ds5 *state,
 	 * TODO: suffix for 2 D457 connected to 1 Deser
 	 */
 	if (state->aggregated & 1)
-		suffix += 4;
+		suffix += 6;
 	snprintf(sd->name, sizeof(sd->name), "D4XX %s %c", name, suffix);
 #else
 	snprintf(sd->name, sizeof(sd->name), "D4XX %s %d-%04x",
@@ -4608,7 +4608,7 @@ static int ds5_mux_init(struct i2c_client *c, struct ds5 *state)
 		 i2c_adapter_id(c->adapter), c->addr);
 #else
 	if (state->aggregated)
-		suffix += 4;
+		suffix += 6;
 	snprintf(sd->name, sizeof(sd->name), "DS5 mux %c", suffix);
 #endif
 	sd->flags |= V4L2_SUBDEV_FL_HAS_DEVNODE;
@@ -5313,7 +5313,7 @@ static int ds5_chrdev_init(struct i2c_client *c, struct ds5 *state)
 	/* Create a device node for this device. */
 #ifndef CONFIG_OF
 	if (state->aggregated)
-		suffix += 4;
+		suffix += 6;
 	snprintf(dev_name, sizeof(dev_name), "%s-%c",
 		DS5_DRIVER_NAME_DFU, suffix);
 #else
-- 
2.25.1

