From da8801112843de4da5309f4e158440c2e801d3d0 Mon Sep 17 00:00:00 2001
From: Florent Pirou <florent.pirou@intel.com>
Date: Wed, 26 Jun 2024 18:18:08 +0800
Subject: [PATCH 22/62] d4xx : fix query sensor info failed on port e

Signed-off-by: Florent Pirou <florent.pirou@intel.com>
---
 drivers/media/i2c/d4xx.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 6a1b04dbe..f796aaa3c 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -3631,6 +3631,10 @@ static int ds5_ctrl_init(struct ds5 *state, int sid)
 		d4xx_controls_q_sub_stream.def = NR_OF_DS5_SUB_STREAMS;
 		d4xx_controls_q_sub_stream.min = NR_OF_DS5_SUB_STREAMS;
 		d4xx_controls_q_sub_stream.max = NR_OF_DS5_SUB_STREAMS * 2 - 1;
+	} else {
+		d4xx_controls_q_sub_stream.def = 0;
+		d4xx_controls_q_sub_stream.min = 0;
+		d4xx_controls_q_sub_stream.max = NR_OF_DS5_SUB_STREAMS - 1;
 	}
 	ctrls->query_sub_stream = v4l2_ctrl_new_custom(hdl, &d4xx_controls_q_sub_stream, sensor);
 
@@ -3718,10 +3722,11 @@ static int ds5_ctrl_init(struct ds5 *state, int sid)
 	default:
 		state->mux.sd.subdev.ctrl_handler = hdl;
 		dev_info(state->mux.sd.subdev.dev,
-			"%s():%d set ctrl_handler for MUX: %p, %s\n",
+			"%s():%d set ctrl_handler for MUX: %p, %s, substream id %d \n",
 			 __func__, __LINE__,
 			 state->mux.sd.subdev.ctrl_handler,
-			 state->mux.sd.subdev.name);
+			 state->mux.sd.subdev.name,
+			 d4xx_controls_q_sub_stream.def);
 		break;
 	}
 
@@ -5384,6 +5389,11 @@ static void ds5_substream_init(struct ds5 *state)
 		state->pad_to_substream[DS5_MUX_PAD_IR]    = 10;
 		state->pad_to_substream[DS5_MUX_PAD_IMU]   = 11;
 	}
+	dev_info(&state->client->dev, "%s() IPU6 CSI2 BE SOC video capture init : \n", __func__);
+	for (i = 0; i < ARRAY_SIZE(state->pad_to_substream); i++)
+	  if (state->pad_to_substream[i] >= 0)
+	    dev_info(&state->client->dev, "pad[%d]->substream=%d\n",
+			 i, state->pad_to_substream[i]);
 	/*
 	 * 0, vc 0, depth
 	 * 1, vc 0, meta data
-- 
2.25.1

