From 69d239eebc3b2a5ab1dd227d36af407bc0f9ec64 Mon Sep 17 00:00:00 2001
From: Florent Pirou <florent.pirou@intel.com>
Date: Thu, 12 Dec 2024 07:57:55 +0000
Subject: [PATCH 26/62] d4xx: crankup V4L2_CID_LINK_FREQ setpoints up 2500Mbps

Signed-off-by: Florent Pirou <florent.pirou@intel.com>
---
 drivers/media/i2c/d4xx.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 8f1f0364a..6920579d8 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -212,6 +212,11 @@ enum ds5_mux_pad {
 	if (max9295_write_8(state, addr, buf)) \
 		return -EINVAL; \
 	}
+#define D4XX_LINK_FREQ_1250MHZ		1250000000ULL
+#define D4XX_LINK_FREQ_1125MHZ		1125000000ULL
+#define D4XX_LINK_FREQ_1000MHZ		1000000000ULL
+#define D4XX_LINK_FREQ_900MHZ		900000000ULL
+#define D4XX_LINK_FREQ_840MHZ		840000000ULL
 #define D4XX_LINK_FREQ_750MHZ		750000000ULL
 #define D4XX_LINK_FREQ_720MHZ		720000000ULL
 #define D4XX_LINK_FREQ_600MHZ		600000000ULL
@@ -333,6 +338,11 @@ static const struct hwm_cmd ewb = {
 };
 #ifdef CONFIG_VIDEO_INTEL_IPU6
 static const s64 link_freq_menu_items[] = {
+	D4XX_LINK_FREQ_1250MHZ,
+	D4XX_LINK_FREQ_1125MHZ,
+	D4XX_LINK_FREQ_1000MHZ,
+	D4XX_LINK_FREQ_900MHZ,
+	D4XX_LINK_FREQ_840MHZ,
 	D4XX_LINK_FREQ_750MHZ,
 	D4XX_LINK_FREQ_720MHZ,
 	D4XX_LINK_FREQ_600MHZ,
@@ -2969,7 +2979,7 @@ static const struct v4l2_ctrl_config d4xx_controls_link_freq = {
 	.max = ARRAY_SIZE(link_freq_menu_items) - 1,
 	.min =  0,
 	.step  = 0,
-	.def = 7,    // default D4XX_LINK_FREQ_300MHZ
+	.def = 12,    // default D4XX_LINK_FREQ_300MHZ
 	.qmenu_int = link_freq_menu_items,
 };
 
-- 
2.25.1

