From 6b2ad531d09f985e9b35e68693457a1a8d358354 Mon Sep 17 00:00:00 2001
From: Josh Watts <jwatts@d3embedded.com>
Date: Fri, 11 Oct 2024 11:52:09 -0400
Subject: [PATCH 34/62] max9295: Force port_sel

Signed-off-by: Florent Pirou <florent.pirou@intel.com>
---
 drivers/media/i2c/max9295.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/i2c/max9295.c b/drivers/media/i2c/max9295.c
index 102a26559..a42f1ac91 100644
--- a/drivers/media/i2c/max9295.c
+++ b/drivers/media/i2c/max9295.c
@@ -252,6 +252,9 @@ int max9295_setup_streaming(struct device *dev)
 			}
 		}
 		dev_dbg(dev, "%s: d4xx specfic setup done\n", __func__);
+	} else {
+	  //HACK: JNW 2024-10-10  Until we actually setup num_streams for proper mapping
+	  port_sel |= 0x61; // Just pipe X & port B
 	}
 
 	for (i = 0; i < g_ctx->num_streams; i++)
-- 
2.25.1

