From 6b4d63594fc2b9ee4942d8106be65c857a6e4391 Mon Sep 17 00:00:00 2001
From: Florent Pirou <florent.pirou@intel.com>
Date: Mon, 13 Jan 2025 23:33:59 +0000
Subject: [PATCH 51/71] lt6911uxc: align linux 6.1x v4l2 uapi

Signed-off-by: Florent Pirou <florent.pirou@intel.com>
---
 drivers/media/i2c/lt6911uxc.c | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/drivers/media/i2c/lt6911uxc.c b/drivers/media/i2c/lt6911uxc.c
index 89ef316ca..2b328ca74 100644
--- a/drivers/media/i2c/lt6911uxc.c
+++ b/drivers/media/i2c/lt6911uxc.c
@@ -462,10 +462,18 @@ static int lt6911uxc_g_input_status(struct v4l2_subdev *sd, u32 *status)
 }
 
 static int __maybe_unused lt6911uxc_s_dv_timings(struct v4l2_subdev *sd,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
+		unsigned int pad,
+#endif
 		struct v4l2_dv_timings *timings)
 {
 	struct lt6911uxc_state *lt6911uxc = to_state(sd);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
+	if (pad != 0)
+		return -EINVAL;
+#endif
+
 	v4l2_dbg(3, debug, sd, "%s():\n", __func__);
 
 	if (!v4l2_valid_dv_timings(timings, lt6911uxc_g_timings_cap(lt6911uxc),
@@ -482,10 +490,18 @@ static int __maybe_unused lt6911uxc_s_dv_timings(struct v4l2_subdev *sd,
 }
 
 static int lt6911uxc_g_dv_timings(struct v4l2_subdev *sd,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
+		unsigned int pad,
+#endif
 		struct v4l2_dv_timings *timings)
 {
 	struct lt6911uxc_state *lt6911uxc = to_state(sd);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
+	if (pad != 0)
+		return -EINVAL;
+#endif
+
 	v4l2_dbg(3, debug, sd, "%s():\n", __func__);
 
 	*timings = lt6911uxc->timings;
@@ -493,10 +509,18 @@ static int lt6911uxc_g_dv_timings(struct v4l2_subdev *sd,
 }
 
 static int __maybe_unused lt6911uxc_query_dv_timings(struct v4l2_subdev *sd,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
+		unsigned int pad,
+#endif
 		struct v4l2_dv_timings *timings)
 {
 	struct lt6911uxc_state *lt6911uxc = to_state(sd);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
+	if (pad != 0)
+		return -EINVAL;
+#endif
+
 	v4l2_dbg(3, debug, sd, "%s():\n", __func__);
 
 	if (false == lt6911uxc->streaming) {
@@ -1132,9 +1156,11 @@ static const struct v4l2_subdev_video_ops lt6911uxc_video_ops = {
 	.g_frame_interval = lt6911uxc_g_frame_interval,
 #endif
 	.g_input_status	= lt6911uxc_g_input_status,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 10, 0)
 //	.s_dv_timings	= lt6911uxc_s_dv_timings,
 	.g_dv_timings	= lt6911uxc_g_dv_timings,
 //	.query_dv_timings	= lt6911uxc_query_dv_timings,
+#endif
 	.s_stream	= lt6911uxc_set_stream,
 };
 
@@ -1147,6 +1173,11 @@ static const struct v4l2_subdev_pad_ops lt6911uxc_pad_ops = {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 8, 0)
 	.get_frame_interval = lt6911uxc_g_frame_interval,
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
+//	.s_dv_timings	= lt6911uxc_s_dv_timings,
+	.g_dv_timings	= lt6911uxc_g_dv_timings,
+//	.query_dv_timings	= lt6911uxc_query_dv_timings,
+#endif
 };
 
 static struct v4l2_subdev_core_ops lt6911uxc_subdev_core_ops = {
-- 
2.25.1

