From 20230436a991044a71bb787df9df999a125e677d Mon Sep 17 00:00:00 2001
From: florent pirou <florent.pirou@intel.com>
Date: Mon, 6 Oct 2025 10:43:10 -0700
Subject: [PATCH 53/71] ar0234: remove gpio irq from linux 6.17

Signed-off-by: florent pirou <florent.pirou@intel.com>
---
 drivers/media/i2c/ar0234.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/ar0234.c b/drivers/media/i2c/ar0234.c
index 90c1af42b..9d653b4fd 100644
--- a/drivers/media/i2c/ar0234.c
+++ b/drivers/media/i2c/ar0234.c
@@ -2378,7 +2378,7 @@ static int ar0234_probe(struct i2c_client *client)
 		dev_err(&client->dev, "failed to init controls: %d", ret);
 		goto probe_error_v4l2_ctrl_handler_free;
 	}
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 17, 0)
 	if (ar0234->platform_data) {
 	    if ((ar0234->platform_data->gpios[0] != -1) && (ar0234->platform_data->irq_pin != -1)) {
 		ret = devm_gpio_request(&client->dev,
@@ -2409,6 +2409,7 @@ static int ar0234_probe(struct i2c_client *client)
 		}
 	    }
 	}
+#endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5, 14, 0)
 	ret = v4l2_async_register_subdev_sensor_common(&ar0234->sd);
 #else
-- 
2.25.1

